packages/excalidraw/data/library.ts

```diff
@@ -74,11 +108,42 @@ export const mergeLibraryItems = (
   return [...newItems, ...localItems];
 };
 
+/**
+ * Returns { deleted, inserted } libraryItems maps, where inserted is
+ * all currently available library items and deleted is all library items
+ * that were deleted since last onLibraryChange call.
+ *
+ * Host apps are recommended to merge `inserted` with whatever state they
+ * have, while removing from the resulting state all items from `deleted`.
+ */
+const createLibraryChange = (
+  prevLibraryItems: LibraryItems,
+  nextLibraryItems: LibraryItems,
+): LibraryChange => {
+  const nextItemsMap = arrayToMap(nextLibraryItems);
+
+  const change: LibraryChange = {
+    deleted: new Map<LibraryItem["id"], LibraryItem>(),
+    inserted: arrayToMap(nextLibraryItems),
```

Comment:
We should unify this with Change introduced in #7348, where inserted isn't all, but really just added / updated items compared to the previous set of all items.

Second part is related to granularity, while in Change we have the deleted / inserted properties of elements, here it's whole elements regardless of the properties. Might be too granular here for now, but would probably aim to do the same here.

Reply:
> We should unify this with Change introduced in #7348, where inserted isn't all, but really just added / updated items compared to the previous set of all items.

True.

> Second part is related to granularity, while in Change we have the deleted / inserted properties of elements, here it's whole elements regardless of the properties. Might be too granular here for now, but would probably aim to do the same here.

Yeah, I've purposely kept this low-granularity as it'd be hard for host apps to handle more granular changes than this (which now doesn't matter since we won't expose it), but also it's simpler and we don't need the added complexity rn.